lib/logstash/outputs/elasticsearch.rb

```diff
@@ -322,10 +322,21 @@ def register
     @bulk_request_metrics = metric.namespace(:bulk_requests)
     @document_level_metrics = metric.namespace(:documents)
 
+    @stop_after_finish_register = Concurrent::AtomicBoolean.new(false)
     @after_successful_connection_thread = after_successful_connection do
       begin
         finish_register
         true # thread.value
+      rescue LogStash::ConfigurationError, LogStash::Outputs::ElasticSearch::HttpClient::Pool::BadResponseCodeError => e
+        if respond_to?(:execution_context) && execution_context.respond_to?(:pipeline_id) &&
+          execution_context.respond_to?(:agent) && execution_context.agent.respond_to?(:stop_pipeline)
```

Comment:
do we need all these checks? is it because during tests we may not have all these available?

Reply:
I see similar checking in [dlq](https://github.com/logstash-plugins/logstash-output-elasticsearch/blob/v11.18.0/lib/logstash/plugin_mixins/elasticsearch/common.rb#L400) and the `stop_pipeline` function is added in recent two years. `pipeline_id` and `agent` are not necessary

Reply:
Tracing back the history of `execution_context.rb` agent and pipeline_id was not there at the [beginning](https://github.com/elastic/logstash/commit/ab1cfe8cf7e20114df58bcc6c996abcb2b0650d7). 
Considering the execution_context was there since 2017, do you think we can safely remove and only check `execution_context.agent.respond_to?(:stop_pipeline)`? I can see other plugins do not check `respond_to?(:execution_context)`

Reply:
@jsvd ☝️